### PR TITLE
Feat(Button): add `xl` and `large` button size variant

### LINKED ISSUE #7003: [button] add `xl` and `large` button size variant
# Background

In the new creation flow pattern we need larger `80px` height [button](https://react.carbondesignsystem.com/?path=/story/button--default). I think we should also add the modal's `64px` buttons to the button component as well for consistency.

**Reasoning:** 
the smaller buttons at the bottom right of the page can easily get lost or overlooked. Having the option for a larger button can add weight and help with usability. It will also stay in proportion to how buttons are treated elsewhere (like modal). 

image

# Task

**Add new button size variants:**
- [x] `height: 80px / 10rem` =  `xl`
- [x] `height: 64px / 8rem` = `large`

# Spec
image

### PR DESCRIPTION
**Pull request by Abdul-Sen**
Closes #7003 

Adds `large` and `xl` variant prop options to the Button component

#### Changelog

**New**

- `lg` and `xl` options for Button `size` prop. Users can now set `size` prop to`<Button size="lg" />` and `<Button size="xl" />` to use large and extra large buttons
- new snapshot for public facing api
- unit tests for the new enhancement
- doc updates to reflect new props

**Changed**

- React Button component
- Button component scss files
- Button component storybook documentation

#### Testing / Reviewing

- Run `yarn test` to verify unit tests are successful for large and xl buttons
- View component in storybook by going to Button component API `size` docs to validate spec requirements as requested in the issue.

### CODE REVIEW

File: packages/react/src/components/Button/Button.js [hunk 1]
```diff
@@ -38,6 +38,8 @@ const Button = React.forwardRef(function Button(
     [`${prefix}--btn`]: true,
     [`${prefix}--btn--field`]: size === 'field',
     [`${prefix}--btn--sm`]: size === 'small' || small,
+    [`${prefix}--btn--lg`]: size === 'large',
```
Comment (tw15egan) [inline packages/react/src/components/Button/Button.js]: Wondering if we should keep this size as `lg` to be consistent with other the other input size conventions, and we can make a note to change `small` to `sm` in our next major release.

**Comment by tw15egan** [comment]
@Abdul-Sen I updated the PR to add the knobs to the Button Playground story so the designers can check it out 👍 

@carbon-design-system/design, should this size variant also apply to icon-only buttons? They look a bit strange

Screen Shot 2020-10-12 at 12 16 43 PM

**Comment by johnbister** [comment]
@tw15egan  I think scaling up the icon for lg and xl is a good solution. Small and field icons look fine as is to me.

I suggest scaling up the icon proportionally to button height.  Let me know if this breakdown makes sense:

Screen Shot 2020-10-14 at 10 29 06 AM

**Comment by aagonzales** [comment]
I talked to @jeanservaas about this one and we concluded that the icon-button variant actually does not need these new sizes. These sizes are for very particular usecases that wouldn't apply to icon-only. 

That being said in the system for buttons we only scale up the icon size if we are also scaling up the type size. So the icon size wouldn't change if we we're keeping this variant.

**Review by johnbister — APPROVED** [review]

File: packages/components/src/components/button/_mixins.scss [hunk 2]
```diff
@@ -104,3 +104,10 @@
     fill: $icon-color;
   }
 }
+
+@mixin button-padding-large {
+  align-items: baseline;
+  padding-top: rem(12px);
+  padding-right: rem(64px);
+  padding-left: rem(12px);
```
Comment (emyarod) [inline packages/components/src/components/button/_mixins.scss]: should we replace these with Carbon spacing tokens?

**Review by emyarod — COMMENTED** [review]
looks good to me, just one note about spacing tokens

File: packages/components/src/components/button/_mixins.scss [hunk 3]
```diff
@@ -104,3 +104,10 @@
     fill: $icon-color;
   }
 }
+
+@mixin button-padding-large {
+  align-items: baseline;
+  padding-top: rem(12px);
```
Comment (andreancardona) [inline packages/components/src/components/button/_mixins.scss]:
```suggestion
  padding-top: $spacing-04;
```

File: packages/components/src/components/button/_mixins.scss (hunk 2 above)
Comment (andreancardona) [inline packages/components/src/components/button/_mixins.scss]:
```suggestion
  padding-left: $spacing-04;
```

**Review by andreancardona — CHANGES_REQUESTED** [review]
@Abdul-Sen

looks good to me - just made some suggestions around spacing tokens - per @emyarod suggestion

File: packages/components/src/components/button/_button.scss [hunk 4]
```diff
@@ -328,6 +328,18 @@
...
+  .#{$prefix}--btn--xl:not(.#{$prefix}--btn--icon-only) {
+    @include button-padding-large;
+
+    min-height: rem(80px);
+  }
+
+  .#{$prefix}--btn--lg:not(.#{$prefix}--btn--icon-only) {
+    @include button-padding-large;
+
+    min-height: rem(64px);
```
Comment (andreancardona) [inline packages/components/src/components/button/_button.scss:340]:
```suggestion
    min-height: $layout-05;
```

File: packages/components/src/components/button/_mixins.scss [hunk 5]
```diff
@@ -104,3 +104,10 @@
...
+  padding-right: rem(64px);
```
Comment (andreancardona) [inline packages/components/src/components/button/_mixins.scss]:
```suggestion
  padding-right: $layout-05;
```

**Comment by Abdul-Sen** [comment]
@andreancardona 

Thanks, committing changes now via GH gui.

File: packages/components/src/components/button/_mixins.scss [hunk 6]
```diff
@@ -104,3 +104,10 @@
     fill: $icon-color;
   }
 }
+
+@mixin button-padding-large {
+  align-items: baseline;
+  padding-top: $spacing-04;
+  padding-right: $layout-05;
+  padding-left: $spacing-04;
```
Comment (aagonzales) [inline packages/components/src/components/button/_mixins.scss]: `padding-left` should be `$spacing-05`

File: packages/components/src/components/button/_mixins.scss [hunk 7]
```diff
@@ -104,3 +104,10 @@
     fill: $icon-color;
   }
 }
+
+@mixin button-padding-large {
+  align-items: baseline;
+  padding-top: $spacing-04;
```
Comment (aagonzales) [inline packages/components/src/components/button/_mixins.scss]: `padding-top` should be `$spacing-05`/16px

**Review by aagonzales — CHANGES_REQUESTED** [review]

File: packages/components/src/components/button/_mixins.scss (hunk 6 above)
Comment (tw15egan) [reply packages/components/src/components/button/_mixins.scss]:
```suggestion
  padding-left: $spacing-05;
```

File: packages/components/src/components/button/_mixins.scss (hunk 7 above)
Comment (tw15egan) [reply packages/components/src/components/button/_mixins.scss]:
```suggestion
  padding-top: $spacing-05;
```

**Review by andreancardona — APPROVED** [review]

**Review by aagonzales — APPROVED** [review]

**Review by emyarod — APPROVED** [review]
looks good to me

**Pull request merged**